File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        j = 0
        maxReach = 0
        for i in range(len(nums)):
            if(not nums[i]):
                if(k == 0):
                    while(nums[j]):
                        j += 1
                    j += 1
                    k += 1
                k -= 1
            maxReach = max(maxReach, i - j + 1)
        return maxReach

File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.go
func longestOnes(nums []int, k int) int {
    var start, res = 0, 0
    for end := range len(nums) {
        if nums[end] == 0 { k-- }
        for k < 0 {
            if nums[start] == 0 {
                k += 1
            }
            start += 1
        }
        
        res = max(res, end - start + 1)
    }
    
    return res
}